addons/delivery/wizard/choose_delivery_package.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class ChooseDeliveryPackage(models.TransientModel):
+    _name = 'choose.delivery.package'
+    _description = 'Choose delivery package type when there is a carrier on the picking'
+
+    stock_quant_package_id = fields.Many2one(
+        'stock.quant.package',
+        default=lambda self: self._default_stock_quant_package_id()
+    ) 
+    delivery_packaging_id = fields.Many2one(
+        'product.packaging',
+        domain=lambda self: self._domain_delivery_packaging_id(),
+        default=lambda self: self._default_delivery_packaging_id()
+    )
+    shipping_weight = fields.Float(
+        string='Shipping Weight',
+        default=lambda self: self._default_shipping_weight()
+    )
+
+    def _default_stock_quant_package_id(self):
+        if self.env.context.get('default_stock_quant_package_id'):
+            return self.env['stock.quant.package'].browse(self.env.context['stock_quant_package_id'])
+
+    def _domain_delivery_packaging_id(self):
+        if self.env.context.get('current_package_carrier_type'):
+            return [('package_carrier_type', '=', self.env.context['current_package_carrier_type'])]
+
+    def _default_delivery_packaging_id(self):
+        res = None
+        if self.env.context.get('default_delivery_packaging_id'):
+            res = self.env['product.packaging'].browse(self.env.context['default_delivery_packaging_id'])
+        if self.env.context.get('default_stock_quant_package_id'):
+            stock_quant_package = self.env['stock.quant.package'].browse(self.env.context['default_stock_quant_package_id'])
+            res = stock_quant_package.packaging_id
+        return res
+
+    def _default_shipping_weight(self):
+        if self.env.context.get('default_stock_quant_package_id'):
+            stock_quant_package = self.env['stock.quant.package'].browse(self.env.context['default_stock_quant_package_id'])
+            return stock_quant_package.shipping_weight
+        else:
+            picking_id = self.env['stock.picking'].browse(self.env.context['active_id'])
+            pack_operation_ids = [po for po in picking_id.pack_operation_ids if po.qty_done > 0 and not po.result_package_id]
```

Comment:
maybe check entire packages also: not x.product_id -> x.package_id.shipping_weight/weight